Introductie ondersteuning voor tab titel slot elementen

stefanborghys opened the pull request:
Er is gekozen voor een implementatie waarbij de 'vl-tabs-pane' een slot kan bevatten voor de titel.
En content blijft op de bestaande manier werken.

```
<vl-tabs-pane data-vl-id="auto">
      <span slot="auto-title-slot"><b>Voorbeeld titel</b></span>
      Voorbeeld content
</vl-tabs-pane>
```

- unit testen aangepast en uitgebreidt
- e2e testen aangepast en uitgebreidt 
- build is groen

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- vl-tabs-pane uitgebreid met een title slot

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- Unit test vl-tabs-pane uitgebreid

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- Unit test vl-tabs aangevuld

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- title slot hidden gemaakt zodat het niet verschijnt als tab content
- title slot wordt verwijdert wanneer het niet gezet is
- linter fix
- tab pane met slot aan de demo pagina toegevoegd
- tab template bouwd een custom template voor een title vanuit het slot, dit onderdeel gaat wel nog fout en moet verder uitgeklaard worden

coemans committed:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- (Tom Coemans) proposal for slotted content

coemans committed:
introduction vl-tabs-pane-title

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- vl-tabs-pane met title-slot toegevoegd aan de demo pagina's
- moveTabPaneTitleSlot methode toegevoegd. Deze cloned het title-slot uit de tabPane naar het bovenliggend tabs niveau. Wordt uitgevoerd tijdens connectedCallback en bij aanpassingen via de reeds bestaande MutationObserver
- opkuis vl-tabs-pane

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- linter fixes

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- aanpassen vl-tabs-pane.test voor een vl-tab-pane met title-slot

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- aanpassen vl-tab.test aan de nieuwe implementatie

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- aanpassen en uitbreiden van de vl-tabs.test aan de nieuwe implementatie
- fix die de originele slotted content verwijdert omdat dit problemen geeft met het opvragen van de innerText van content (zie test: er kunnen dynamisch tabs toegevoegd worden)

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- aanpassen en uitbreiden van de tabs.test aan de nieuwe implementatie

stefanborghys commented on demo/vl-tabs-base.html:
```diff
@@ -28,8 +28,12 @@
       <vl-tabs-pane data-vl-id="fiets" data-vl-title="Fiets">
         Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
       </vl-tabs-pane>
+      <vl-tabs-pane data-vl-id="auto" data-vl-title="Trein">
```
data-vl-title mag weg

stefanborghys commented on src/vl-tabs-all.js:
```diff
@@ -1,11 +1,13 @@
 import {VlTab} from '/src/vl-tab.js';
 import {VlTabs} from '/src/vl-tabs.js';
 import {VlTabsPane} from '/src/vl-tabs-pane.js';
+import {VlTabsPaneTitle} from '/src/vl-tabs-pane-title.js';
```
mag verwijdert worden

stefanborghys commented on src/vl-tabs-pane-title.js:
```diff
@@ -0,0 +1,30 @@
+ * @see {@link https://webcomponenten.omgeving.vlaanderen.be/demo/vl-tabs.html|Demo}
+ *
+ */
+export class VlTabsPaneTitle extends vlElement(HTMLElement) {
```
niet meer nodig

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR fixes na een eerste eigen controle

SanderKleykens commented on src/vl-tabs.js:
```diff
@@ -100,7 +104,7 @@ export class VlTabs extends vlElement(HTMLElement) {
     return this._template(`
       <li is="vl-tab" data-vl-href="${pathname}#${id}" data-vl-id="${id}">
-        ${(title)}
+        <slot name="${id}-title-slot">${(title)}</slot>
```
Past dit slot niet beter in de vl-tab-pane component? Dan moet je ook niet met `${id}` prefixen werken in de slot name.
De suffix kan in principe ook weg, aangezien je normaal wel weet dat het over een slot gaat als gebruiker.

stefanborghys replied:
Dit slot komt opgevuld uit de vl-tab-pane indien een slotted tag is voorzien anders wordt de titel uit het attribuut genomen en in deze slot gestopt. 
De vl-tab die je hier ziet zet dan weer wat de vl-tab-pane aanbied om in een titel in de vorm van een link.

pascalriquier commented on demo/vl-tabs-active-tab.html:
```diff
@@ -28,9 +28,13 @@
         <vl-tabs-pane data-vl-id="fiets" data-vl-title="Fiets">
           Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
         </vl-tabs-pane>
+        <vl-tabs-pane data-vl-id="auto">
+          <span slot="auto-title-slot"><b>Auto</b></span>
+          Duis vitae magna vitae eros pretium porttitor id quis mauris. Sed imperdiet a diam in suscipit. Nunc consectetur urna nunc, eu tempor odio rutrum non. Sed in sem convallis, placerat nisi nec, placerat velit. In fringilla ex sed malesuada dictum. Sed congue neque orci, quis porta mi tempus ultrices. Vivamus gravida magna eu enim aliquet, sit amet tempus ante pellentesque. Donec eget hend ...
+        </vl-tabs-pane>
```
Denk dat de gewoonte is om per "feature" een demo component te maken, dus ik zou hier overwegen om een aparte "vl-tabs met titles als slots" te maken.

stefanborghys replied:
- nieuwe demo pagina toegevoegd
- vl-tabs-pane verwijdert

pascalriquier commented on demo/vl-tabs-base.html:
```diff
@@ -28,8 +28,12 @@
       <vl-tabs-pane data-vl-id="fiets" data-vl-title="Fiets">
         Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
       </vl-tabs-pane>
+      <vl-tabs-pane data-vl-id="auto">
+        <span slot="auto-title-slot"><b>Auto</b></span>
+        Duis vitae magna vitae eros pretium porttitor id quis mauris. Sed imperdiet a diam in suscipit. Nunc consectetur urna nunc, eu tempor odio rutrum non. Sed in sem convallis, placerat nisi nec, placerat velit. In fringilla ex sed malesuada dictum. Sed congue neque orci, quis porta mi tempus ultrices. Vivamus gravida magna eu enim aliquet, sit amet tempus ante pellentesque. Donec eget hendre ...
+      </vl-tabs-pane>
```
Als je dus apart een demo maakt met slot-titles heb je deze hier niet nodig.

stefanborghys replied:
- nieuwe demo pagina toegevoegd
- vl-tabs-pane verwijdert

pascalriquier commented on demo/vl-tabs.html:
```diff
@@ -27,6 +26,10 @@
       <vl-tabs-pane data-vl-id="fiets" data-vl-title="Fiets">
         Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
       </vl-tabs-pane>
+      <vl-tabs-pane data-vl-id="auto">
+        <span slot="auto-title-slot"><b>Auto</b></span>
+        Duis vitae magna vitae eros pretium porttitor id quis mauris. Sed imperdiet a diam in suscipit. Nunc consectetur urna nunc, eu tempor odio rutrum non. Sed in sem convallis, placerat nisi nec, placerat velit. In fringilla ex sed malesuada dictum. Sed congue neque orci, quis porta mi tempus ultrices. Vivamus gravida magna eu enim aliquet, sit amet tempus ante pellentesque. Donec eget hendre ...
+      </vl-tabs-pane>
```
Idem opmerking vl-tabs-base

stefanborghys replied:
heb de tab met title slot toch behouden in deze vl-tabs.html demo omdat deze ook gebruikt wordt op de demo pagina van het departement omgeving (de andere demo pagina's zie ik daar niet). Heb wel een extra test case geschreven die specifiek de vl-tabs-title-slot.html pagina test

pascalriquier replied:
als je daar deze mee bedoelt 
https://webcomponenten.omgeving.vlaanderen.be/demo/vl-tabs.html

die wordt gepubliceerd van zodra er een nieuwe release komt. dus die zal bevatten wat jij er hier nu van maakt ...

pascalriquier commented on src/vl-tabs.js:
```diff
@@ -187,6 +203,7 @@ export class VlTabs extends vlElement(HTMLElement) {
     const tabPanesToAdd = mutations.flatMap((mutation) => [...mutation.addedNodes]).filter((node) => node instanceof VlTabsPane);
     tabPanesToAdd.forEach((tabPane) => {
       const index = this.__tabPanes.indexOf(tabPane);
+      this._moveTabPaneTitleSlot(tabPane);
       this._addTab({id: tabPane.id, title: tabPane.title, index: index});
```
Denk duplicatie hier van het feit dat je altijd moveTabPaneTitleSlot moet doen als je addTab doet?

stefanborghys replied:
deduplicatie _moveTabPaneTitleSlot methode in vl-tabs. Deze wordt nu enkel opgeroepen in de _addTab methode

pascalriquier commented on test/e2e/components/vl-tab.js:
```diff
@@ -4,7 +4,18 @@ const {By} = require('vl-ui-core').Test.Setup;
+    await clickableElement.click();
+  }
+
+  async getTitleText() {
+    const link = await this._getLink();
+    const slot = await link.findElement(By.css('slot'));
+    const slotElements = await this.getAssignedElements(slot);
+    const titleElement = slotElements.length === 0 ? slot : slotElements[0];
```
Duplicatie

stefanborghys replied:
gededupliceerd

pascalriquier commented on test/unit/vl-tabs.test.html:
```diff
@@ -160,4 +183,4 @@
     });
   </script>
```
misschien nog een explicitie test toevoegen dat alles ook correct werkt indien er aan het title slot iets veranderd wordt? maw dat die change "doorgepropageerd" wordt.

stefanborghys replied:
test bijgeschreven die nagaat of het updaten van een title slot doorkomt. En dat was niet het geval. De mutator keek niet naar de elementen binnen de vl-tabs-pane (enkel de rechtstreekse). En er werd ook nog geen rekening gehouden om het gekopiërde slot te vervangen door de nieuwe inhoud (enkel toevoegen was nu het geval).

pascalriquier reviewed (changes requested):
zie opmerkingen

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie, demo/vl-tab-title-slot.html toegevoegd
- PR suggestie, vl-tabs-pane met title slot verwijdert uit demo/vl-tabs-base.html
- PR suggestie, vl-tabs-pane met title slot verwijdert uit demo/vl-tabs-active-tab.html
- PR suggestie, vl-tab.js test component code gededupliceerd
- PR suggestie, e2e tab.test.js test cases aangepast

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- test content verwijdert
- linter fix

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie, deduplicatie _moveTabPaneTitleSlot methode in vl-tabs. Deze wordt nu enkel opgeroepen in de _addTab methode

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie, test bijgeschreven die nagaat of het updaten van een title slot doorkomt. En dat was niet het geval. De mutator keek niet naar de elementen binnen de vl-tabs-pane (enkel de rechtstreekse). En er werd ook nog geen rekening gehouden om het gekopiërde slot te vergangen door de nieuwe inhoud (enkel toevoegen was nu het geval).
- opkuis test en verbetering unit testen

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie, heb de tab met title slot toch behouden in de vl-tabs.html demo omdat deze ook gebruikt wordt op de demo pagina van het departement omgeving (de andere demo pagina's zie ik daar niet). Heb wel een extra test case geschreven die specifiek de vl-tabs-title-slot.html pagina test

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- extra Tab met title slot demo voorzien op de vl-tabs.html pagina

stefanborghys commented:
- PR suggesties doorgenomen en code aangepast
- hertest + groene build
- klaar om te herreviewen

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie: tabs.test.js slot test aangepast zodat deze niet meer een eigen demo pagina gebruikt. Overbodige bestanden en text verwijdert

stefanborghys reviewed (commented):
Double check

stefanborghys commented:
- aanpassing doorgevoerd
- build groen

stefanborghys commented:
@pascalriquier bekijk je nog even mijn laatste wijziging van gisteren

pascalriquier commented on src/vl-tabs.js:
```diff
@@ -195,23 +205,33 @@ export class VlTabs extends vlElement(HTMLElement) {
+
+    const tabPanesToUpdate = mutations.flatMap((mutation) => [...mutation.addedNodes]).filter((node) => node.parentNode instanceof VlTabsPane).map((node) => node.parentNode);
+    tabPanesToUpdate.forEach((tabPane) => {
+      const index = this.__tabPanes.indexOf(tabPane);
+      this._removeTab(tabPane.id);
+      this._addTab({tabPane: tabPane, index: index});
+
+      this._removeTabSection(tabPane.id);
+      this._addTabSection({id: tabPane.id, index: index});
```
Dit lijkt op "removeTab" en dan "addTab".
En ik zie hier een beetje boven ook addTab en addTabSection, wat hier dan ook weer staat, dus lijkt mij nog wat duplicatie.

stefanborghys replied:
deduplicatie van add, remove van tab en tabSection aangezien deze meermaals worden aangeroepen. Bij deze toegevoegd.

pascalriquier reviewed (changes requested):
zie opmerking

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie: deduplicatie van add, remove van tab en tabSection aangezien deze meermaals worden aangeroepen

stefanborghys commented:
- laatste aanpassing doorgevoerd
- build groen

stefanborghys commented:
@coemans deze staat klaar ter review voor jou

coemans commented:
@stefanborghys ik ga deze PR verder verwerken, maar zie dat er momenteel 2 conflicten zijn. Kan jij deze even bekijken? Thanks!

coemans commented on demo/vl-tabs.html:
```diff
@@ -30,6 +29,19 @@
+      <vl-tabs-pane data-vl-id="vrachtwagen">
+        <span slot="vrachtwagen-title-slot"><i>Vrachtwagen</i></span>
+        Duis vitae magna vitae eros pretium porttitor id quis mauris. Sed imperdiet a diam in suscipit. Nunc consectetur urna nunc, eu tempor odio rutrum non. Sed in sem convallis, placerat nisi nec, placerat velit. In fringilla ex sed malesuada dictum. Sed congue neque orci, quis porta mi tempus ultrices. Vivamus gravida magna eu enim aliquet, sit amet tempus ante pellentesque. Donec eget hendre ...
+      </vl-tabs-pane>
```
De `<u/>` en `<i/>` elementen hebben geen effect dus heb ik verwijderd.

stefanborghys replied:
ok

coemans committed:
removed style tags

coemans commented on demo/vl-tabs.html:
```diff
@@ -30,6 +29,19 @@
+  <vl-demo data-vl-title="Tabs met title slot">
+    <vl-tabs id="tabs-slotted">
+      <vl-tabs-pane data-vl-id="auto">
+        <span slot="auto-title-slot"><b>Auto</b></span>
```
Het lijkt me beter om `slot="title"` te gebruiken als API. Indien het nodig is om specifieker het slot te definiëren kan dit best automatisch gebeuren zoals in het voorbeeld bij branch #slotted-content.

stefanborghys replied:
ok, kan dit aanpassen zie geen impact op de code momenteel. Nu er was een reden waarom ik dat zo expliciet genoemd heb ik vermoed dat ik het ergens in andere code ook zo gezien had maar ben daarvan niet meer zeker.

stefanborghys replied:
slot naam aangepast

coemans commented on demo/vl-tabs.html:
```diff
@@ -30,6 +29,19 @@
+  <vl-demo data-vl-title="Tabs met title slot">
+    <vl-tabs id="tabs-slotted">
+      <vl-tabs-pane data-vl-id="auto">
+        <span slot="auto-title-slot">Auto</span>
+        Duis mollis, est non commodo luctus, nisi erat porttitor ligula.
+      </vl-tabs-pane>
+      <vl-tabs-pane data-vl-id="vrachtwagen">
+        <span slot="vrachtwagen-title-slot">Vrachtwagen</span>
+        Duis vitae magna vitae eros pretium porttitor id quis mauris. Sed imperdiet a diam in suscipit. Nunc consectetur urna nunc, eu tempor odio rutrum non. Sed in sem convallis, placerat nisi nec, placerat velit. In fringilla ex sed malesuada dictum. Sed congue neque orci, quis porta mi tempus ultrices. Vivamus gravida magna eu enim aliquet, sit amet tempus ante pellentesque. Donec eget hendre ...
+      </vl-tabs-pane>
```
Het lijkt me beter om `slot="title"` te hanteren als API. Is het niet mogelijk het slot dynamisch te onderscheiden zodat afnemers niet verplicht worden om het id te verwerken in de slot waarde.

stefanborghys replied:
idem, antwoord bovenstaande. Lijkt me dat dit wel kan aangepast worden naar title.

coemans commented on src/vl-tabs.js:
```diff
@@ -146,9 +153,28 @@ export class VlTabs extends vlElement(HTMLElement) {
+    if (tabPaneTitleSlot) {
+      const clonedTabPaneTitleSlot = tabPaneTitleSlot.cloneNode(true);
+
+      const existingTabPaneTitleSlot = this.querySelector(`vl-tabs > [slot="${tabPane.id}-title-slot"]`);
+      if (existingTabPaneTitleSlot) {
+        this.replaceChild(clonedTabPaneTitleSlot, existingTabPaneTitleSlot);
```
Ik vermoed dat deze code nooit uitgevoerd wordt aangezien de selector nooit resultaat geeft. De code is bijgevolg overbodig?

stefanborghys replied:
als ik de code debug kom ik hier weldegelijk in. Dit stuk is toegevoegd geweest om wijzigingen die in het tabsPane title slot gebeuren te laten doorkomen. Als we enkel adden komt bij elke wijziging er gewoon een nieuwe title bij. Door de replace vangen we dat op. Zie test/unit/vl-tabs.test.html

coemans commented on src/vl-tabs.js:
```diff
@@ -146,9 +153,28 @@ export class VlTabs extends vlElement(HTMLElement) {
+        this.appendChild(clonedTabPaneTitleSlot);
+      }
+
+      // We verwijderen het src title slot, anders zal de innerText van de tabsection deze nog bevatten
+      tabPaneTitleSlot.remove();
```
Deze code lijkt me overbodig? De code zorgt er ook voor dat afnemers nooit het slot element kunnen wijzigen.

stefanborghys replied:
ja zoals verduidelijkt in de comment, dit is bewust gedaan om een side-effect waarbij je de innerText van het tabSection slot bevraagd de title  text niet mee te krijgen ( zie comment op de test code, ' test/unit/vl-tabs.test.html'). Nu als dit geen probleem is kan dit voor mij weg.

stefanborghys replied:
tab pane title slot wordt niet meer verwijdert

coemans commented on src/vl-tabs.js:
```diff
@@ -179,24 +205,38 @@ export class VlTabs extends vlElement(HTMLElement) {
-      this._removeTabSection(tabPane.id);
+    tabPanesToDelete.forEach((tabPane) => this.__removeTabAndSection(tabPane));
+
+    const tabPanesToUpdate = mutations.flatMap((mutation) => [...mutation.addedNodes])
+        .filter((node) => node.parentNode instanceof VlTabsPane)
+        .map((node) => node.parentNode);
+    tabPanesToUpdate.forEach((tabPane) => {
+      this.__removeTabAndSection(tabPane);
+      this.__addTabAndSection(tabPane);
     });
```
Deze code lijkt overbodig?

stefanborghys replied:
deze is in functie van 'test/unit/vl-tabs.test.html', 'tabs pane title slot wijzigingen updaten de tab' toegevoegd. Nu ik zag net het commentje van dit stukje code de test niet deed falen. Ik heb dit ondertussen aangepast. Te bekijken dus of die functionaliteit nodig is.

coemans replied:
Ik kan nog steeds deze code verwijderen zonder dat er unit of e2e testen falen, dus lijkt de code overbodig te zijn of er ontbreken vermoedelijk nog testen om aan te tonen waarvoor deze code nodig is?

stefanborghys replied:
dit stukje wordt gebruikt om de title dynamisch te updaten. De code loopt hier door als ik de vl-tabs unit testen draai. Ik zal nog een test case aanmaken.

stefanborghys replied:
ik zie nu net dat hier wel een unit test voor bestaat met naam 'tabs pane title slot wijzigingen updaten de tab' uit vl-tabs.test.html. Als ik dan weer de code in comment zet faalt die test weldegelijk.

![afbeelding](https://user-images.githubusercontent.com/1375279/118462743-4d9d4400-b6ff-11eb-80e5-e84f24c97b6f.png)

Firefox:
<img width="1552" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/118462789-5726ac00-b6ff-11eb-8307-18a6e75b7ebe.png">

Chrome:
<img width="1312" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/118462880-6efe3000-b6ff-11eb-8b47-49666b36de55.png">

stefanborghys replied:
Als ik de code dan weer uncomment: 

![afbeelding](https://user-images.githubusercontent.com/1375279/118463035-98b75700-b6ff-11eb-85af-5587c46cfed0.png)

Wordt die test groen. 
Firefox:
<img width="1552" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/118463095-a836a000-b6ff-11eb-8ecb-84585e1edd8a.png">

Chrome: 
<img width="1312" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/118463177-c69c9b80-b6ff-11eb-8e21-a93768777c86.png">

stefanborghys replied:
heb ook het gevoel dat we dit al besproken hadden tijdends de laatste conf call enige tijd geleden. Nu ja wat doe ik hier nu weer mee. Nog een testje schrijven, de demo uitbreiden, ...

coemans replied:
Ik heb vermoedelijk lokaal niet lang genoeg gewacht, want ik had voor de zekerheid de code in commentaar gecommit. Bamboo toont aan dat er wel een unit test faalt, dus de code is inderdaad nodig en blijkt ook uit de unit testen.

coemans replied:
Ik heb nog een code refactoring uitgevoerd.

coemans commented:
@stefanborghys ik heb de PR gedeeltelijk verwerkt maar al enkele opmerkingen/vragen. Misschien best deze eerst even (eventueel samen) bekijken voordat ik de review verder afwerk.

stefanborghys commented on test/unit/vl-tabs.test.html:
```diff
@@ -126,9 +138,15 @@
+                assert.lengthOf(getTabSectionArray(), 4);
+                assert.deepEqual(getTabArray().map((tab) => tab.id), ['1', '2', '3', '4']);
+                assert.deepEqual(getTabSectionArray().map((tabSection) => tabSection.querySelector('slot').assignedNodes()[0].innerText), ['1 content',
+                  '2 content', '3 content', '4 content']);
```
@coemans 
als we in src/vl-tabs.js `tabPaneTitleSlot.remove();` verwijderen. Zal deze test falen de innerText van de 4de tabsectie zal dan '44 content' verwachten.
Dit klopt de text van het title slot meegenomen wordt. Willen we dit side-effect? Anders mag voor mij die remove er ook uit.

stefanborghys replied:
test case aangepast nadat het slot niet meer verwijdert wordt

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- test fix

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie: slot naam aanpassen naar 'title'

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie: remove weggehaald en test case aangepast

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie: .only verwijdert

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie: test case aangepast t.g.v. het niet meer verwijderen van het slot

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie: querySelector vereenvoudigt

Commit:
Merge remote-tracking branch 'origin/master' into UIG-1210-slot-voor-tab-title

# Conflicts:
#	src/vl-tabs.js
#	test/unit/vl-tabs.test.html

stefanborghys commented:
@coemans heb alle besproken comments opgelost evenals de merge conflicten.
Deze kan dus nog een gereviewed worden.

coemans committed:
build

coemans committed:
fixed endless loop triggered by refresh of unit page

coemans committed:
commented code to indicate no tests fail

coemans committed:
revert code

coemans committed:
refactoring

coemans committed:
page object refactor and added slotted node support

Commit:
#25 - UIG-1210 - Slot voorzien voor de tab title:
- PR suggestie: test case aangepaste, code aangepast

Commit:
Merge remote-tracking branch 'origin/UIG-1210-slot-voor-tab-title' into UIG-1210-slot-voor-tab-title

# Conflicts:
#	src/vl-tabs.js

coemans committed:
revert

Commit:
Merge remote-tracking branch 'origin/UIG-1210-slot-voor-tab-title' into UIG-1210-slot-voor-tab-title

# Conflicts:
#	src/vl-tabs.js

coemans committed:
revert unit tests

coemans commented on test/unit/vl-tabs-pane.test.html:
```diff
@@ -26,8 +35,13 @@
+      test('title null wanneer een title-slot wordt gebruikt', () => {
+        const tabPane = fixture('vl-tab-pane-met-title-slot-fixture');
+        assert.isNull(tabPane.title);
+      });
```
Hier wordt uitsluitend getest dat het data-vl-title attribuut niet gezet werd. Ik heb deze test verwijderd omdat deze me overbodig lijkt.

coemans committed:
removed unit test

coemans committed:
refactoring

coemans commented on test/unit/vl-tabs.test.html:
```diff
@@ -59,14 +77,22 @@
+
+        const tabTitleSlot = element.shadowRoot.querySelector(`slot[name='${autoTabsPane.id}-title-slot']`);
+        const tabTitleSpan = tabTitleSlot.assignedNodes()[0];
+        assert.equal(tabTitleSpan.innerHTML, '<b>Auto</b>');
```
Ik heb hiervoor een nieuwe unit test geïntroduceerd.

coemans committed:
unit test refactoring

coemans committed:
refactoring

coemans committed:
fix e2e tests
